components/weekly_storage/weekly_storage.cc

```diff
@@ -76,6 +76,26 @@ uint64_t WeeklyStorage::GetWeeklySum() const {
                          });
 }
 
+uint64_t WeeklyStorage::GetHighestValueInWeek() const {
+  // We record only value for last N days.
+  const base::Time n_days_ago =
+      clock_->Now() - base::TimeDelta::FromDays(kDaysInWeek);
+  std::list<DailyValue> last_weeks_daily_values(daily_values_.size());
+  auto copied_it =
+      std::copy_if(daily_values_.begin(), daily_values_.end(),
```

Comment:
looks like we can do it in one pass without making a new vector? Just by finding maximum among values for which `i.day > n_days_ago`

Reply:
Since max is strictly comparing left vs right (and doing less than)... what should be returned if the date range is invalid?

Reply:
probably `end()`?

Reply:
left as-is for now, but I can write a test program and figure it out if you'd like (I'm just not confident in doing it in one go without writing a test program)